src/csharp/Microsoft.Spark/Interop/Ipc/JsonSerDe.cs

```diff
@@ -2,70 +2,158 @@
+        /// <summary>
+        /// Helper method to Parse an object into a JsonElement.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public static JsonElement Parse(object obj)
+        {
+            return Parse(JsonSerializer.SerializeToUtf8Bytes(obj));
         }
 
         /// <summary>
-        /// Extend method to sort items in a JSON array by keys.
+        /// Helper method to Parse a byte array in UTF8 to a JsonElement.
+        /// </summary>
+        private static JsonElement Parse(byte[] utf8Json)
+        {
+            using (var document = JsonDocument.Parse(utf8Json))
+            {
+                return document.RootElement.Clone();
+            }
+        }        
+
+        /// <summary>
+        /// Sort all types of Json properties.
+        /// </summary>
+        private static void SortPropertiesCore(this JsonElement jsonElement, Utf8JsonWriter writer)
+        {
+            switch (jsonElement.ValueKind)
+            {
+                case JsonValueKind.Undefined:
+                    throw new InvalidOperationException();
+                case JsonValueKind.Object:
+                    jsonElement.SortObjectProperties(writer);
+                    break;
+                case JsonValueKind.Array:
+                    jsonElement.SortArrayProperties(writer);
+                    break;
+                case JsonValueKind.String:
+                case JsonValueKind.Number:
+                case JsonValueKind.True:
+                case JsonValueKind.False:
+                case JsonValueKind.Null:
+                    jsonElement.WriteTo(writer);
+                    break;
+            };
+        }
+
+        /// <summary>
+        /// Sort properties that are Json objects.
         /// </summary>
-        public static JArray SortProperties(this JArray jArray)
+        private static void SortObjectProperties(this JsonElement jObject, Utf8JsonWriter writer)
         {
-            if (jArray.Count == 0)
+            Debug.Assert(jObject.ValueKind == JsonValueKind.Object);
+
+            var propertyNames = new List<string>();
+            foreach (JsonProperty prop in jObject.EnumerateObject())
             {
-                return jArray;
+                propertyNames.Add(prop.Name);
             }
+            propertyNames.Sort();
 
-            var sortedJArray = new JArray();
-            foreach (JToken item in jArray)
+            writer.WriteStartObject();
+            foreach (string name in propertyNames)
             {
-                if (item is JObject elem)
-                {
-                    sortedJArray.Add(elem.SortProperties());
-                }
-                else if (item is JArray arrayElem)
-                {
-                    sortedJArray.Add(arrayElem.SortProperties());
-                }
+                writer.WritePropertyName(name);
+                jObject.GetProperty(name).WriteElementHelper(writer, true);
```

Comment:
nit: Use named parameters
```suggestion
                jObject.GetProperty(name).WriteElementHelper(writer, writePrimitives: true);
```